src/applications/static-pages/subscribeComponentAnalyticsEvents.js

```diff
@@ -0,0 +1,50 @@
+import recordEvent from 'platform/monitoring/record-event';
+
+const analyticsEvents = [
+  {
+    componentName: 'Modal',
+    prefix: 'modal',
+    actions: [{ action: 'show', name: 'int-modal-click' }],
+  },
+  {
+    componentName: 'AdditionalInfo',
+    prefix: 'additional-info',
+    actions: [
+      { action: 'expand', name: 'int-additional-info-expand' },
+      { action: 'collapse', name: 'int-additional-info-collapse' },
+    ],
+  },
+];
+
+export default function subscribeComponentAnalyticsEvents() {
+  document.body.addEventListener('component-library-analytics', e => {
+    // Is it a component we are tracking?
+    const component = analyticsEvents.find(
+      c => c.componentName === e.detail.componentName,
+    );
+
+    // Is it an action we are tracking?
+    if (component && component.actions) {
+      const action = component.actions.find(
+        ev => ev.action === e.detail.action,
+      );
+
+      if (action) {
+        const dataLayer = { event: action.name };
```

Comment:
Why do we need this `action` object in `analyticsEvents`? Why don't we just get the "event" based on the event we are handling, rather than constructing a new data structure so that we can look things up in?

Reply:
Are you asking why we don't just pass in `int-modal-click` instead of `show`?

Reply:
I'm fine with the `<Modal>` having its action be `show` - I think that makes sense. Right now it isn't obvious to me why we are translating this to `int-modal-click`. Is that due to how our GA is set up?

Reply:
> Is that due to how our GA is set up?

That is correct.  The `event` param in the dataLayer is what we use in GTM to trigger our GA (and other) tags.

![image](https://user-images.githubusercontent.com/62304138/108878392-38b48700-75ce-11eb-90e4-f27dac546f95.png)

Reply:
Gotcha. This makes sense.